### PR TITLE
singleproductpage

### PR DESCRIPTION
**Pull request by preethasri**
created single product page and added some responsiveness to webpage

Screenshot (14)

### CODE REVIEW

**Review by bipin7yadav — APPROVED** [review]
code looks clean

File: src/context/index.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
...
-export {AuthProvider} from './AuthContext'
-export {WishListProvider} from './WishListContext'
-export {CartProvider} from './CartContext'
\ No newline at end of file
+export {CategoryProvider,useCategory} from './category-context'
+export {FilterProvider,useFilter} from './filter-context'
+export {ProductProvider,useProduct} from './product-context'
+export {AuthProvider,useAuth} from './AuthContext'
+export {WishListProvider,useWishList} from './WishListContext'
+export {CartProvider,useCart} from './CartContext'
```
Comment (someshtiwarii) [inline src/context/index.js:8]: i like the barrell export here

File: src/pages/cartmanagement/cartmanagement.css [hunk 2]
```diff
@@ -34,12 +34,13 @@
   
   .img-and-details{
     display:flex;
-  width:800px;
+  max-width:800px;
```
Comment (someshtiwarii) [inline src/pages/cartmanagement/cartmanagement.css:37]: can use rem here

File: src/pages/productpage/allproducts/AllProducts.js [hunk 3]
```diff
@@ -27,7 +28,7 @@ export  const AllProducts = ({ product }) => {
     wishList.find(wishlistProduct => wishlistProduct._id === productId);
       
     const isInCart=productId=>cart.find(cartProduct =>cartProduct._id ===productId)
-                
+       console.log(_id)
```
Comment (someshtiwarii) [inline src/pages/productpage/allproducts/AllProducts.js:31]: can remove console logs

**Review by someshtiwarii — APPROVED** [review]
clean UI! looks good to merge!

**Pull request merged**